SCAutolib/models/local_ca.py

```diff
@@ -0,0 +1,144 @@
+from pathlib import Path
+from shutil import rmtree, copy
+
+from SCAutolib import TEMPLATES_DIR, run, logger
+from SCAutolib.models.ca import CA
+
+
+class LocalCA(CA):
+    template = Path(TEMPLATES_DIR, "ca.cnf")
+
+    def __init__(self, root_dir: Path = Path("/etc/SCAutolib/ca")):
+        """
+        Class for local CA. Initialize required attributes, real setup is made
+        by LocalCA.setup() method
+
+        :param root_dir: Path to root directory of the CA. By default, is in
+                         /etc/SCAutolib/ca
+        :type: Path
+        """
+        self.root_dir: Path = root_dir
+        self._conf_dir: Path = Path(root_dir, "conf")
+        self._newcerts: Path = Path(root_dir, "newcerts")
+        self._certs: Path = Path(root_dir, "certs")
+        self._crl: Path = Path(root_dir, "crl", "root.crl")
+        self._ca_pki_db: Path = Path("/etc/sssd/pki/sssd_auth_ca_db.pem")
+
+        self._ca_cnf: Path = Path(self._conf_dir, "ca.cnf")
+        self._ca_cert: Path = Path(root_dir, "rootCA.pem")
+        self._ca_key: Path = Path(root_dir, "rootCA.key")
+
+        self._serial: Path = Path(root_dir, "serial")
+        self._index: Path = Path(root_dir, "index.txt")
+
+    def setup(self, force: bool = False):
+        """
+        Creates directory and file structure needed by local CA. If directory
+        already exists and force = True, directory would be recursively deleted
+        and new local CA would be created. Otherwise, configuration would be
+        skipped.
+
+        :param force: overwrite existing configuration with force if True,
+                      otherwise, skip configuration.
+        """
+        if self.root_dir.exists():
+            logger.warning(f"Directory {self.root_dir} already exists.")
+            if not force:
+                logger.warning("Skipping configuration.")
+                return
+            logger.warning("Removing configuration.")
+            rmtree(self.root_dir)
+
+        self.root_dir.mkdir(parents=True, exist_ok=True)
+        self._ca_cnf.parent.mkdir()
+        self._newcerts.mkdir()
+        self._certs.mkdir()
+        self._crl.parent.mkdir()
+
+        # Copy template and edit it with current root dir for CA
+        copy(self.template, self._ca_cnf)
+        with self._ca_cnf.open("r+") as f:
+            f.write(f.read().format(ROOT_DIR=self.root_dir))
+        with self._serial.open("w") as f:
+            f.write("01")
+
+        self._index.touch()
```

Comment:
We can probably also provide empty file as a template, but its not a big deal if we create it here also

Reply:
Similar to the previous comment, do not see any reason to create any auxiliary files (more over, empty files). This would make the library just bigger and the code more complex (need to copy something somewhere vs directly creat the file where it has to be)